packages/strapi-plugin-users-permissions/admin/src/containers/AuthPage/index.js

```diff
@@ -40,73 +41,99 @@ import styles from './styles.scss';
-      switch (this.props.match.params.authType) {
+    if (authType !== prevProps.match.params.authType) {
+      const params = search ? replace(search, '?code=', '') : id;
+      setForm(authType, params);
+      hideLoginErrorsInput(false);
+    }
+    if (submitSuccess && submitSuccess !== prevProps.submitSuccess) {
+      switch (authType) {
         case 'login':
-        case 'reset-password':
-          this.props.history.push('/');
+        case 'reset-password': 
+          auth.getToken()
+            ? this.redirect('/')
+            : this.redirect('/plugins/users-permissions/auth/login');
           break;
         case 'register':
-          this.props.history.push('/');
+          this.redirect('/');
           // NOTE: prepare for comfirm email;
-          // this.props.history.push(`/plugins/users-permissions/auth/register-success/${this.props.modifiedData.email}`);
+          // this.redirect(`/plugins/users-permissions/auth/register-success/${this.props.modifiedData.email}`);
           break;
         default:
-
       }
     }
   }
 
+  // Get form Errors shortcut.
+  getFormErrors = () => {
+    const { formErrors } = this.props;
+    return get(formErrors, ['0', 'errors', '0', 'id']);
+  }
+
   handleSubmit = (e) => {
+    const { modifiedData, setErrors, submit } = this.props;
     e.preventDefault();
-    const formErrors = Object.keys(this.props.modifiedData).reduce((acc, key) => {
-      if (isEmpty(get(this.props.modifiedData, key)) && !isBoolean(get(this.props.modifiedData, key))) {
+    const formErrors = Object.keys(modifiedData).reduce((acc, key) => {
+      if (isEmpty(get(modifiedData, key)) && !isBoolean(get(modifiedData, key))) {
         acc.push({ name: key, errors: [{ id: 'components.Input.error.validation.required' }] });
       }
 
-      if (!isEmpty(get(this.props.modifiedData, 'password')) && !isEmpty(get(this.props.modifiedData, 'confirmPassword')) && findIndex(acc, ['name', 'confirmPassword']) === -1) {
-        if (this.props.modifiedData.password.length < 6) {
+      if (!isEmpty(get(modifiedData, 'password')) && !isEmpty(get(modifiedData, 'confirmPassword')) && findIndex(acc, ['name', 'confirmPassword']) === -1) {
+        if (modifiedData.password.length < 6) {
           acc.push({ name: 'password', errors: [{ id: 'users-permissions.components.Input.error.password.length' }] });
         }
         
-        if (get(this.props.modifiedData, 'password') !== get(this.props.modifiedData, 'confirmPassword')) {
+        if (get(modifiedData, 'password') !== get(modifiedData, 'confirmPassword')) {
           acc.push({ name: 'confirmPassword', errors: [{ id: 'users-permissions.components.Input.error.password.noMatch' }] });
         }
       }
 
       return acc;
     }, []);
 
-    this.props.setErrors(formErrors);
+    setErrors(formErrors);
 
     if (isEmpty(formErrors)) {
-      this.props.submit(this.context);
+      submit(this.context);
     }
   }
 
+  isAuthType = type => {
```

Comment:
I think a more explicit naming like `getAuthType` would be more understandable than `isAuthType`